### PR TITLE
Split core.go file into logical subparts

### LINKED ISSUE #215: Split core.go into logical subparts
The file internal/core/core.go is getting close to 1k lines and contains several types of function definitions. The length and combination of functions makes it harder than it has to be to find the dependencies between functions. 
Candidates for separations can be: 
- Function calls directly related to API endpoint requests. 
- Attribute related functions
- Parallelisation of the attribute request
- More simple support functions

Currently it is unclear how the import of the c code bundle these functions.

### PR DESCRIPTION
**Pull request by AdilKhurshid**
Splitting the primary core.go file into more logical sub-sections which will facilitate in easier code maintenance and readability in the future.

Closes #215

### CODE REVIEW

File: internal/core/core_fence.go
Comment (achaikou) [inline internal/core/core_fence.go:8]: Didn't think that we would have to include all these in every file :pouting_cat: 

Wondering what is better - leave it as is or always include only capi.h, as it in itself actually has import of ctypes and stdlib. Probably would depend on compiler though if it compiles or not.

So I would probably vote for what you currently have, but would gladly hear other opinions :)

File: internal/core/core_fence.go
Comment (achaikou) [inline internal/core/core_fence.go:5]: I think we cam get rid of these in slice, fence and attributes files and leave them in core.go one only? At least it still compiles if I remove these two lines.

File: internal/core/core.go
Comment (achaikou) [inline internal/core/core.go:230]: Maybe we should leave it in core? This is the only one "translation" method that gets moved out. Not regular surface or bounds.

So maybe it would be easier to find here?

File: internal/core/core.go
Comment (yngve793) [inline internal/core/core.go:1]: Can we move "type RegularSurface struct" and "cRegularSurface" to core_attributes.go? 
It seams like they are only used in this file.

Comment (yngve793) [inline internal/core/core.go:1]: Specific metadata types like SliceMetadata can probably be moved to corresponding file.

**Review by yngve793 — COMMENTED** [review]
Try to keep shared code in the remaining core.go.

File: internal/core/core_attributes.go
Comment (AdilKhurshid) [reply internal/core/core_attributes.go:454]: Sure

File: internal/core/core.go
Comment (AdilKhurshid) [reply internal/core/core.go:230]: Sure

File: internal/core/core.go
Comment (AdilKhurshid) [reply internal/core/core.go:1]: Yes, we can move those struct as well. Good suggestion

**Review by achaikou — APPROVED** [review]
I am happy :cat: As for me - rebase and merge, but lets wait for the second reviewer's stamp of approval and agreement :)

**Review by yngve793 — APPROVED** [review]
No further comments.

**Pull request merged**
